elastic/kibana
# [Synthetics] project monitor  - delete

## Summary

Relates to https://github.com/elastic/kibana/issues/142418

Adds a route to delete all given monitors for the specified project

`DELETE /api/synthetics/project/{projectName}/monitors/_bulk_delete`

Accepts the following schema, which monitors as an array of journey ids for the monitors to be deleted
```
{
    monitors: ['joureyId1', journeyId2']
}
```

Returns the following schema, with a list of successfully deleted monitors
```
{
   deleted_monitors: ['joureyId1', journeyId2']
}
```

Accepts a max of 250 monitors at a time. Returns an error when the payload is too large. 

Returns an error when the user does not have permission to delete monitors with private locations

### Testing

Create a group of project monitors as normal
1. Navigate to Kibana. Generate an api key
2. Create a new project using the synthetics-agent, or reuse a new old one
3. Navigate to that project, run `SYNTHETICS_API_KEY={theykey} npm run push`
4. post to new delete api to delete monitors`curl -X DELETE http://{username}:{password}@{basepath}/api/synthetics/project/{projectName}/monitors -H 'kbn-xsrf: true -d '{“monitors”: [“yourJorneyIdsToTestWith”, “yoursecondJourneyIdToTestWith”]}'
` Make sure to url-encode your project name
5. You should see a response with your monitors. Example
```
{
   deleted_monitors: ['joureyId1', journeyId2']
}
```
